### PR TITLE
Topo flux

### PR DESCRIPTION
**Pull request by NimrodCarmon**
Topographic expansion to the forward model. Considers surface geometry and adjusts the downwelling fluxes (direct and diffuse) accordingly.

### CODE REVIEW

File: isofit/configs/sections/input_config.py [hunk 1]
```diff
@@ -72,6 +72,13 @@ def __init__(self, sub_configdic: dict = None):
         str: Input 'loc', or location, file.  Can be either a .mat, .txt, or ENVI formatted binary cube.
         Provides per-pixel lat, long, and elevation information.
         """
+
+        self._surface_geometry_file_type = str
+        self.surface_geometry_file = None
+        """
+        str: Input 'loc', or location, file.  Can be either a .mat, .txt, or ENVI formatted binary cube.
+        Provides per-pixel lat, long, and elevation information. xxx surface geometry file
```
Comment (pgbrodrick) [inline isofit/configs/sections/input_config.py:80]: Please update.

File: isofit/configs/sections/radiative_transfer_config.py [hunk 2]
```diff
@@ -232,6 +232,8 @@ def __init__(self, sub_configdic: dict = None):
         self.lut_grid = None
 
         self._unknowns_type = RadiativeTransferUnknownsConfig
+        # This is type annotation syntax. Basically assigns "None" to the variable, but
+        # adds the class to the __annotations__ attribute, so other things could check
```
Comment (pgbrodrick) [inline isofit/configs/sections/radiative_transfer_config.py:236]: Please remove.

**Review by pgbrodrick — COMMENTED** [review]
Thanks @NimrodCarmon, this is looking pretty good.  I need to finish a some tests and a more comprehensive review, but for now I'll suggest:

- handling the various notes given inline
- we're starting to get pretty data-heavy on this repository, and I'm going to recommend migrating the example data (radiance, obs, and supporting modtran runs) to a public-access server and just using a curl to download (similar to the most recent cube example).  I can do this and send over a PR if you like, given that we don't yet have a contribution guide along these lines.

I don't love continuing to expand the geometry object (especially with things that aren't necessary geometry), but I've also done this on other branches.  Ultimately we need a different data object, but that's for a larger refactor, so I say we let this go for now.

File: isofit/core/isofit.py [hunk 3]
```diff
@@ -65,21 +63,23 @@ def __init__(self, config_file, row_column='', level='INFO', logfile=None):
         self.config.get_config_errors()
 
         # Initialize ray for parallel execution
-        rayargs = {'address': self.config.implementation.ip_head,
+        rayargs = {#self.config.implementation.ip_head, ! this is so isofit works with vpn on
                    '_redis_password': self.config.implementation.redis_password,
                    'ignore_reinit_error': self.config.implementation.ray_ignore_reinit_error,
                    'local_mode': self.config.implementation.n_cores == 1}
-
+
```
Comment (pgbrodrick) [inline isofit/core/isofit.py:70]: Please revert

File: isofit/core/isofit.py [hunk 4]
```diff
@@ -65,21 +63,23 @@ def __init__(self, config_file, row_column='', level='INFO', logfile=None):
...
             rayargs['_temp_dir'] = self.config.implementation.ray_temp_dir
             # Used to run on a VPN
-            ray.services.get_node_ip_address = lambda: '127.0.0.1'
+            #ray.services.get_node_ip_address = lambda: '127.0.0.1'
 
         # We can only set the num_cpus if running on a single-node
         if self.config.implementation.ip_head is None and self.config.implementation.redis_password is None:
             rayargs['num_cpus'] = self.config.implementation.n_cores
+        _node_ip_address = '192.168.1.247' # the changes here are so we can run with vpn on
+        ray._private.services.address_to_ip = lambda _node_ip_address: _node_ip_address
```
Comment (pgbrodrick) [inline isofit/core/isofit.py:82]: Please revert.

**Comment by davidraythompson** [comment]
Superseded by #264, closing at the request of @NimrodCarmon

**Pull request closed without merging**